src/mesh/impls/bout/boutmesh.cxx

```diff
@@ -2014,57 +2014,58 @@ void BoutMesh::clear_handles() {
  *                   Communication utilities
  ****************************************************************/
 
-int BoutMesh::pack_data(const vector<FieldData*> &var_list, int xge, int xlt, int yge, int ylt, BoutReal *buffer) {
-  int jx, jy, jz;
-  int len = 0;
-  std::vector<FieldData*>::iterator it;
+int BoutMesh::pack_data(const vector<FieldData *> &var_list, int xge, int xlt, int yge,
+                        int ylt, BoutReal *buffer) {
 
-  for(jx=xge; jx != xlt; jx++) {
+  int len = 0;
 
+  for (int jx = xge; jx != xlt; jx++) {
     /// Loop over variables
-    for(const auto& var : var_list) {
-      if(var->is3D()) {
+    for (const auto &var : var_list) {
+      if (var->is3D()) {
         // 3D variable
-
-        for(jy=yge;jy < ylt;jy++)
-          for(jz=0;jz < LocalNz;jz++)
-            len += var->getData(jx,jy,jz,buffer+len);
+        for (int jy = yge; jy < ylt; jy++) {
+          for (int jz = 0; jz < LocalNz; jz++, len++) {
+            buffer[len] = (*dynamic_cast<Field3D*>(var))(jx, jy, jz);
+          }
+        }
       } else {
         // 2D variable
-        for(jy=yge;jy < ylt;jy++)
-          len += var->getData(jx,jy,0,buffer+len);
+        for (int jy = yge; jy < ylt; jy++, len++) {
+          buffer[len] = (*dynamic_cast<Field2D*>(var))(jx, jy);
+        }
       }
     }
   }
 
-  return(len);
+  return (len);
 }
 
-int BoutMesh::unpack_data(const vector<FieldData*> &var_list, int xge, int xlt, int yge, int ylt, BoutReal *buffer)
-{
-  int jx, jy, jz;
-  int len = 0;
-  std::vector<FieldData*>::iterator it;
+int BoutMesh::unpack_data(const vector<FieldData *> &var_list, int xge, int xlt, int yge,
```

Comment:
Oh, you are right. I was thinking why we need the dynamic cast, but that is due to field data being possibly of another type ...
Sorry for the confusion :-/